### ISSUE TITLE
No Cydia on multiple devices

### ISSUE THREAD

**Issue by iM4CH3T3** [iabem97/topanga#1]
Have ran it now and nothing there. Not even the bootstrap folder was there. No cydia application also.

**Comment by nullpixel** [comment]
We need more information to be able to debug this. Do you have logs to hand? Please upload them to a website such as ghostbin, so we are able to take a look.

**Comment by Counterassy** [comment]
hi, it's probaply do to line 266 in jailbreak.m
seems like a development leftover that installs cydia but only if it's already there ;)

**Comment by iM4CH3T3** [comment]
How to get the log ?

**Comment by zhaoyabei** [comment]
@Counterassy 
As you said,I change file jailbreak.m (line 266):
` if(should_install_cydia != YES) `
to:
`if(should_install_cydia == YES) `
and cydia is install success.

Hope it help @iabem97

**Comment by TimVanDenEnden** [comment]
zhaoyabei your right it install cydia (on iPhone X iOS 11.1.2), 
when you change it to:
` if(should_install_cydia == YES)`

but it when you try to open cydia crash after 3 to 5 seconds.

Hope this helps as well.

**Comment by TimVanDenEnden** [comment]
Forgot to add the log :P 

https://ghostbin.com/paste/hcejd/raw

**Comment by TimVanDenEnden** [comment]
Thx Counterassy i will try that later today.

**Comment by TimVanDenEnden** [comment]
Counterassy I search in his (nullpixel) repositrories but i can not find the branch can you give the link?

**Comment by Counterassy** [comment]
no no, the branch is created of this master. You should find it here, it is just called nullpixel/feature

**Comment by TimVanDenEnden** [comment]
Thx found the branch i am gone look around after i am done with working thx Counterassy.

**Comment by iM4CH3T3** [comment]
Here’s the log 

https://www.dropbox.com/s/qakaomsf9nxjq5q/topanga-2017-12-23-062323.ips?dl=0

**Comment by nullpixel** [comment]
Don’t use my branch

**Comment by TimVanDenEnden** [comment]
Can you explain why nullpixel? (Trying to understand what everthing does)

**Comment by nullpixel** [comment]
It’s outdated

**Comment by a3ms** [comment]
it says WTF no binlist?

**Comment by iM4CH3T3** [comment]
Works with lates build...!!!
Cydia Installer, Cydia Translations, UIKitTools and Patcyh are showing as partially installed and have this red like on screenshot

**Comment by iM4CH3T3** [comment]
https://imgur.com/gallery/vYqMz

**Comment by a3ms** [comment]
https://m.imgur.com/W4C69A1?r

**Comment by TimVanDenEnden** [comment]
Guys can you tell me what you changed?

**Comment by TimVanDenEnden** [comment]
because its still not running on my iPhone x the cydia app keeps crashing :(

**Comment by Counterassy** [comment]
look here:
https://github.com/iabem97/topanga/issues/9

**Comment by TimVanDenEnden** [comment]
does it then run?

**Comment by Counterassy** [comment]
give it a try for your self

**Comment by TimVanDenEnden** [comment]
https://imgur.com/gallery/lHPlF

Really nice man thx can you explain me what this sleep(3) did?

**Comment by Counterassy** [comment]
well it sleeps (i guess for 3sec) but thats not really the point.

it‘s ment as some kind of error handling i guess, but I have no idea why :)

**Comment by TimVanDenEnden** [comment]
that make kind of sense if you reader the function name :P 

I also found out cydia will only run when the other app is running in the background think it using some cache thing.

**Comment by SteBowker10** [comment]
Wouldn’t recommend changing output from default 3 seconds otherwise output rk32 tfp0 times out causing a kernel panic. Yes it’s a workaround but not the best of fixes.

**Comment by Counterassy** [comment]
the craton of a seperate ther is commented just re-enable it and it should work

**Comment by iabem97** [comment]
just change the if statement. I won't be changing it since I already have Cydia installed and need it to always untar for testing.
TL;DR: if you do not have Cydia already installed, just change the if statement in jailbreak.m from:
`if(should_install_cydia != YES) {` to `if(should_install_cydia == YES) {`

**Comment by ACEWRITER** [comment]
No cydia on IPhone 8+
